### DISCUSSION TITLE
Bevy relicensing

### DISCUSSION THREAD

**Discussion by TheRawMeatball** [bevyengine/bevy#2358] [General]
We are currently under just the MIT license ['for simplicity'](https://discord.com/channels/691052431525675048/749335865876021248/771851125116502056). However, several core contributors feel that we can do better, specifically by using the [recommended](https://rust-lang.github.io/api-guidelines/necessities.html#crate-and-its-dependencies-have-a-permissive-license-c-permissive) combination of either MIT or Apache 2.0. This license combination is the same license combination used by the [rust compiler](https://github.com/rust-lang/rust) itself.

This would be good for integration with the rest of the ecosystem, as well as helping avoid the need for (non-GPL) end user projects to repeat the MIT license text in full once for every bevy crate in their graph. An example of the sort of integration which might currently be prevented by the license would be if the rust book wanted to include a bevy based example.

If there is a reason you are unwilling to relicense your contributions, please make a top level discussion. (The terms of the Apache 2.0 are broadly in the same class as those of the MIT license, so I'd be surprised if any of these come up.)

If you think we should relicense, please react to this with :+1:

**Comment by Nilirad** [comment]
Better late than ever. While Bevy has more than 200 contributors, those who added very significant contributions are way less. If all of them agree, we can relicense easily even if we don't get approval from every single contributor.

On the other hand, if we wait to the point where a core contributor becomes unreachable, it will become impossible to switch without rewriting significant portions of the engine.

**Comment by recatek** [comment]
Per discussion on discord just now, I'd also like to put MIT-0 up for consideration as a third option (aside from MIT and Apache 2.0). It's the same as MIT, just without the line about attribution: https://opensource.org/licenses/MIT-0
